wordpress-mobile/WordPress-iOS
# [Stats Refresh] No Results view when the Insights Overview fails

Fixes #10379 

This PR adds the no results screen when the whole overview fails.

## Store
I followed the logic the `StatsInsightsStore` uses when the api service is fetching a particular overview section. I use different `Bool` value types, each for every overview section. Then the viewModel and the view controller will check only the `fetchingOverviewHasFailed` property.
Also `func tableViewModel() -> ImmuTable` will return always an empty `ImmuTable` if `fetchingOverviewHasFailed` will be true.


## To test:
• From My Sites select a site and open its dashboard
• Activate your _Network Link Conditioner_ or any other tool you want to use. The important thing is the response should be always **100% Loss**
• Select Stats to open the Stats section. Be sure the tab Insights is selected.
• You should see the loading view (if the tool you are using allows the service to try a connection) and then the no results screen with the _Retry_ button
• Disable your _Network Link Conditioner_
• Tap on Retry button and leave the service load the overview.
• Enable again your _Network Link Conditioner_ and pull down to refresh
• You should see the again the no results screen.

**Important:** **DO NOT SWITCH OFF** your network because the reachability will automatically display the no internet connection screen.


Update release notes:
